pkg/crypto/prf/prf.go

```diff
@@ -74,6 +74,24 @@ func PSKPreMasterSecret(psk []byte) []byte {
 	return out
 }
 
+// EcdhePSKPreMasterSecret implements TLS 1.2 Premaster Secret generation given a psk, a keypair and a curve
+func EcdhePSKPreMasterSecret(psk, publicKey, privateKey []byte, curve elliptic.Curve) ([]byte, error) {
+	preMasterSecret, err := PreMasterSecret(publicKey, privateKey, curve)
+	if err != nil {
+		return nil, err
+	}
+	pskEnc := make([]byte, 2+len(preMasterSecret)+2+len(psk))
```

Comment:
Can you add some space to these operations? Maybe put allocations first, and then do copies? Somewhat harder to follow with no empty lines.

These are subjective, so feel free to push back.
* Use offsets instead of two slices. Users that don't know Go as well can be confused about reference vs value.

Reply:
fixed